ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/ForkIdManager.java

```diff
@@ -66,12 +69,41 @@ public ForkIdManager(final Blockchain blockchain, final List<Long> forks) {
               blockchain,
               genesisHash,
               forks,
-              fs -> fs.stream().distinct().collect(Collectors.toUnmodifiableList()),
+              fs -> fs.stream().distinct().sorted().collect(Collectors.toUnmodifiableList()),
               new ArrayList<>());
       this.forkIDCheckers = Arrays.asList(newForkIdChecker, legacyForkIdChecker);
     }
   }
 
+  public ForkId computeForkId() {
+    final long head = chainHeadSupplier.getAsLong();
+    if (lastHead != 0 && head == lastHead && lastComputedForkId != null) {
+      return lastComputedForkId;
+    }
+    lastHead = head;
+
+    final CRC32 crc = new CRC32();
+    long next = 0;
+    crc.update(genesisHash.toArray());
+    for (final Long fork : forks) {
+      if (fork <= head) {
+        updateCrc(crc, fork);
+        continue;
+      }
+      next = fork;
+      break;
+    }
+    lastComputedForkId = new ForkId(getCurrentCrcHash(crc), next);
+    return lastComputedForkId;
+  }
+
+  ForkId getLatestForkId() {
```

Comment:
Is this just for testing?

Reply:
Yes. Added `@VisibleForTesting` annotation.